random1on1/api/channels.py

```diff
@@ -1,36 +1,132 @@
 
-class WriteOnlyChannel(ABC, TextChannel):
+        self.channel = channel
 
     @abstractmethod
-    def __init__(self, channel_name: str):
-        # TODO: Add preconditions here
-        self.channel_name = channel_name
+    async def set_permissions(self, default_role: Role, random1on1_role: Role):
+        yield
 
 
-class WriteOnlyDirectMessages(ABC, GroupChannel):
+class AnnouncementChannel(Random1on1Channel):
 
-    @abstractmethod
-    def __init__(
-        self,
-        channel_name: str,
-        participants: list[Participant],
-    ):
-        # TODO: Add proconditions here
-        self.channel_name = channel_name
-        self.participants = participants
+    def __init__(self, name: str, category: CategoryChannel):
+        super().__init__(name, category)
+        _ = self.send_opening_announcement()
+
+    async def send_opening_announcement(self):
+        messages = await self.channel.history(limit=1).flatten()
+        if len(messages) == 0:
+            self.channel.send("""
+                Hello Fellow Humans!
+
+                My name is the Random 1 on 1 bot! I will be running a program where you (if you choose to opt in) will be matched randomly every week
+                with a new person from the discord server. The idea is to promote social interaction by getting everyone to meet new people from the
+                server they might not have talked to before. I will send over an introduction to kick off a conversation, and then you and your 
+                partner for the week can take it from there. It's recommended that you spend at least 15 minutes for a chat (either over discord, 
+                or video conference or, if applicable, in person), but there's not hard and fast rules. The point of the program is just to have fun!
+
+                Have fun!
+
+                - Random 1 on 1 bot
+            """)
+
+    async def set_permissions(self, default_role: Role, random1on1_role: Role):
+        _ = self.channel.set_permissions(default_role,
+                                         read_messages=True,
+                                         write_messages=False)
+        _ = self.channel.set_permissions(random1on1_role,
+                                         read_messages=True,
+                                         write_messages=True)
+
+    async def announce_pairings(self, pairings: Pairings):
+        announcement_message = f"""
+        Announcing the pairings for Random 1 on 1s week of {datetime.now().strftime('%Y-%m-%d')}:
+        ---
+        """
+        for component in connected_components(pairings.pairing_graph):
+            announcement_message += "/".join(
+                [participant.name for participant in component]) + "\n"
+        _ = self.channel.send(announcement_message)
+
+
+class HistoryChannel(Random1on1Channel):
+
+    def __init__(self, name: str, category: CategoryChannel):
+        super().__init__(name, category)
```

Comment:
I think this two lines of code are not necessary. The init method is already defined in the parent class.

Reply:
I'll double check that it's not necessary -- it might be due to ABC's `@abstractmethod` label, but we could probably just remove that from the parent class if needed.